module1/index.md

```diff
@@ -2,9 +2,72 @@
 
 How to build a hello world application on nextjs and deploy it using Vercel:
 
-## Deployment
+## Context
+
+In this module you will be learning about some of the web development technologies that we are most excited about. You will go through the basic concepts, understanding the technologies we are using in this workshop and why, we will then take you through how to get started in building your first hello world application and how to deploy and get it running.
+
+Let's begin with some of the basic concepts about tools and technologies we are using in this workshop:
+
+### Client Side Rendering vs Server Side Rendering
+
+Recently Server side rendering has been getting more and more attractions among the web development community and that is thanks to React and its built-in server-side hydration features. Before we begin, let's just first step back to understand the difference. The main difference between the two is where the web content is rendered. On CSR approach the page content is rendered on the client browser using java scripts for dynamic rendering vs on SSR approach the content is fully rendered on the server and sent to the client.
+
+Why should you use Server Side Rendering?
+
+1. Server-side rendering delivers fully rendered HTML to the client, which leads to faster initial page loads compared to client-side rendering.
+2. Search engines typically have difficulty parsing and indexing client-rendered JavaScript content. With server-side rendering, the initial HTML is sent to the client, making it easier for search engine crawlers to read and index the content. This can positively impact the visibility and ranking of your website in search engine results.
+3. SSR enables code sharing between the client and the server, reducing duplication and improving  efficiency. It also simplifies the development by having to maintain one codebase for both frontend and backend.
+4. Server-side rendering can provide a better perceived performance. Users see fully rendered content immediately upon loading the page, reducing the time it takes to see meaningful content and interact with the application.
+5. Server-side rendering can be beneficial for users on low-end devices or with limited network connectivity. By offloading the rendering process to the server, the client device requires less processing power and memory, resulting in improved performance and reduced battery consumption.
```

Comment:
This ChatGPT?

Reply:
It's a mix of researching through some websites and some help from chatgpt